02-express-tutorial/controllers/people.js

```diff
@@ -0,0 +1,68 @@
+const { people } = require("../data")
+
+const getPeople = (req, res) => {
+    res.status(200).json(people)
+}
+
+const addPerson = (req, res) => {
+    if (!req.body.name) {
+        return res.status(400).json({ success: false, message: "Please provide a name" });
+    }
+    people.push({ id: people.length + 1, name: req.body.name });
+    res.status(201).json({ success: true, name: req.body.name });
+}
+
+const getPerson = (req, res) => {
+
+    const checkPerson = checkPersonId(req, res)
+
+    if (!checkPerson.success) {
+        return res.status(404).send(checkPerson)
+    }
+
+    res.send(checkPerson.person )
+}
+
+const updatePerson = (req, res) => {
+
+    const checkPerson = checkPersonId(req, res)
+
+    if (!checkPerson.success) {
+        return res.status(404).send(checkPerson)
+    }
+
+    const newPerson = people.map((person) => {
+        if (person.id === Number(req.params.id)) {
+            person.name = req.body.name
+        }
+        return person
+    })
+
+    res.status(200).json({ success: true, data: checkPerson.person, people })
```

Comment:
This variable is unused. Instead of using the `map` method, you could use the `find` method to update the user and include the updated version in your response. Also, I don't think you should include the complete list in the response, just the resource that was impacted by the request.

```suggestion
    const person = people.find(p => p.id === Number(req.params.id));
    person.name = req.body.name;
    res.status(200).json({ success: true, data: person });
```